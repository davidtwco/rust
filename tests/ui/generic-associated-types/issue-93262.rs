//@ check-fail

pub trait Trait {
    type Assoc<'a> where Self: 'a;
}

pub trait Foo<T: Trait>
where
    for<'a> T::Assoc<'a>: Clone
{}

pub struct Type;

impl<T: Trait> Foo<T> for Type //~ ERROR: the parameter type `T` may not live long enough
where
    for<'a> T::Assoc<'a>: Clone
{}

fn main() {}
