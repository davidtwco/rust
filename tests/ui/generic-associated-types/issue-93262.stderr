error[E0311]: the parameter type `T` may not live long enough
  --> $DIR/issue-93262.rs:14:16
   |
LL | impl<T: Trait> Foo<T> for Type
   |                ^^^^^^
   |                |
   |                the parameter type `T` must be valid for lifetime `'a`...
   |                ...so that the type `T` will meet its required lifetime bounds...
   |
note: ...that is required by this bound
  --> $DIR/issue-93262.rs:9:27
   |
LL |     for<'a> T::Assoc<'a>: Clone
   |                           ^^^^^
   = help: consider adding an explicit lifetime bound `T: 'a`...

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0311`.
